compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -450,13 +453,13 @@ public void visit(BLangResourceFunction funcNode) {
 
     @Override
     public void visit(BLangFunction funcNode) {
+        validateParams(funcNode);
```

Comment:
Shouldn't we validate the return type?

Reply:
It is validated in `logErrorsForInferredArrays`.